CASSGO-57 Deprecate Session.ExecuteBatch() and move "execute batch" methods to Batch type

tengu-alt opened the pull request:
This PR deprecates `Session.ExecuteBatch()` and others similar methods(`MapExecuteBatchCAS`,`ExecuteBatchCAS`)

worryg0d reviewed (approved):
LGTM

tengu-alt commented:
@joao-r-reis Could you please take a look?

joao-r-reis commented on session.go:
```diff
@@ -788,6 +790,29 @@ func (s *Session) ExecuteBatchCAS(batch *Batch, dest ...interface{}) (applied bo
+// was sent.
+// Further scans on the interator must also remember to include
+// the applied boolean as the first argument to *Iter.Scan
+func (b *Batch) ExecCAS(dest ...interface{}) (applied bool, iter *Iter, err error) {
```
these methods are all duplicated, we can make the new ones call the old ones or vice versa

tengu-alt replied:
Agree, fixed.

tengu-alt committed:
Deprecate Session.ExecuteBatch() and move "execute batch" methods to Batch type

Session.ExecuteBatch() was deprecated and similar methods(MapExecuteBatchCAS,ExecuteBatchCAS)
were deprecated and moved to Batch type

patch by Oleksandr Luzhniy; reviewed by João Reis, Bohdan Siryk, for CASSGO-57

joao-r-reis commented:
Looking good, can you resolve the conflicts? I think it's because I just merged the vector PR sorry. After that it's good to be merged.

Oh also please add the changelog entry

joao-r-reis commented:
And format the commit message

tengu-alt commented:
@joao-r-reis Done!
